spec/spec_helper.rb

```diff
@@ -101,25 +103,119 @@
 # rubocop:disable Metrics/MethodLength
 # rubocop:disable Metrics/PerceivedComplexity
 def generate_test_image(file, params = {})
-  width = params[:width] || '100'
-  height = params[:height] || '100'
-  color = params[:color] || 'TrueColor'
-  profile = params[:profile] || 'sRGBIEC6196621'
-  image_type = params[:image_type]
-  create_command = "convert rose: -scale #{width}x#{height}\! -type #{color} "
-  create_command += ' -profile ' + File.join(Assembly::PATH_TO_IMAGE_GEM, 'profiles', profile + '.icc') + ' ' unless profile == ''
-  create_command += " -type #{image_type} " if image_type
-  create_command += ' -compress lzw ' if params[:compress]
-  create_command += file
-  create_command += ' 2>&1'
-  output = `#{create_command}`
-  raise "Failed to create test image #{file} (#{params}): \n#{output}" unless $CHILD_STATUS.success?
+  # Set default size for sample test image.
+  line_size = 1
+  box_size = 6
+  width = (box_size * 6) + (line_size * 7)
+  height = (box_size * 5) + (line_size * 6)
+
+  # Set parameters for image creation options.
+  image_type  = params[:image_type] || File.extname(file)
+  bands       = params[:bands] || 3
+  color       = params[:color] || 'rgb'
+  depth       = params[:depth] || 8
+  cg_type     = params[:cg_type] || 'adobe_rgb'
+  compression = params[:compression]
+  profile     = params[:profile]
+
+  temp_array = color_gauge_values(cg_type)
+  temp_image = Vips::Image.black(width, height, bands: temp_array.first.size)
+  (0..4).each do |i|
+    b = (box_size * i) + (line_size * (i + 1))
+    # d = b + box_size - line_size
+    (0...6).each do |j|
+      a = (box_size * j) + (line_size * (j + 1))
+      # c = a + box_size - line_size
+      colors = temp_array.shift
+      temp_image = temp_image.draw_rect(colors, a, b, box_size, box_size, fill: true)
+    end
+  end
+
+  temp_image = color.eql?('cmyk') ? temp_image.copy(interpretation: :cmyk) : temp_image.copy(interpretation: :srgb)
+
+  temp_image = if color.eql?('grey') && bands == 3
+                 mean = temp_image.bandmean
+                 Vips::Image.bandjoin([mean, mean, mean])
+               elsif color.eql?('grey') && bands == 1
+                 temp_image.bandmean
+               elsif color.eql?('bin') && bands == 3
+                 mean = temp_image.bandmean < 128
+                 Vips::Image.bandjoin([mean, mean, mean])
+               elsif color.eql?('bin') && bands == 1
+                 temp_image.bandmean < 128
+               else
+                 temp_image
+               end
+
+  options = {}
+  unless profile.nil?
+    profile_file = File.join(TEST_PROFILE_DIR, profile + '.icc')
+    options.merge!(profile: profile_file)
+  end
+
+  case image_type
+  when '.tiff', '.tif'
+    options.merge!(compression: compression) unless compression.nil?
+    options.merge!(squash: true) if depth.eql?(1)
+    temp_image.tiffsave(file, **options)
+  when '.jpeg', '.jpg', '.jp2'
+    temp_image.jpegsave(file, **options)
+  end
 end
```

Comment:
use `jpegsave` for saving jp2s as well (just for test images)

Reply:
This actually ends up creating a .jpeg image with a .jp2 file extension. It doesn't really change the use in the spec, though.

Reply:
please add a comment explaining this (why `jpegsave` instead of `jp2ksave` is used here and that it is creating a .jpeg image with a jp2 file extenstion, but that's okay for testing purposes)

Reply:
that's an old github comment, that line of code to create JP2s is gone and we are not creating Jp2 test images anymore for tests (wasn't really needed)

Reply:
I meant the comments in this PR review, not in the code.  Your comment and Tony's both give valuable context about calling `.jpegsave` that is likely useful to code maintainers.